### PR TITLE
rainfuck-cli: init at 1.0.0

### PR DESCRIPTION
**Pull request by IogaMaster**
https://gitlab.com/Oglo12/rainfuck-cli

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ra/rainfuck-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    owner = "Oglo12";
+    repo = "rainfuck-cli";
+    rev = "dc9a885ef7daa4e4a60133c6114e582378978a85";
+    hash = "sha256-hQ4xfwii6P3YphxQAwzY9wraNIiae9vyuLSEXq5xL6I=";
+  };
+
+  cargoHash = "sha256-n/ZWMEjSzyOBC9EVtGi9/13GnttCA5oqWlRcqan5gic=";
+
+  meta = with lib; {
+    description = "";
```
Comment (dotlambda) [inline pkgs/by-name/ra/rainfuck-cli/package.nix:20]:
```suggestion
    description = "Run Brainfuck in the terminal";
```
?

File: pkgs/by-name/ra/rainfuck-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, rustPlatform
+, fetchFromGitLab
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rainfuck-cli";
+  version = "unstable-2024-01-20";
```
Comment (dotlambda) [inline pkgs/by-name/ra/rainfuck-cli/package.nix:8]: I guess this is supposed to be 1.0.0? https://gitlab.com/Oglo12/rainfuck-cli/-/commit/dc9a885ef7daa4e4a60133c6114e582378978a85
Maybe ask upstream.

Comment (IogaMaster) [reply pkgs/by-name/ra/rainfuck-cli/package.nix:8]: Okay

Comment (dotlambda) [reply pkgs/by-name/ra/rainfuck-cli/package.nix]: > Maybe ask upstream.

Did you? Upstream should create a tag.

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/ra/rainfuck-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "rainfuck-cli";
+  version = "1.0.0";
+
+  src = fetchFromGitLab {
+    owner = "Oglo12";
+    repo = "rainfuck-cli";
+    rev = "dc9a885ef7daa4e4a60133c6114e582378978a85";
+    hash = "sha256-hQ4xfwii6P3YphxQAwzY9wraNIiae9vyuLSEXq5xL6I=";
```
Comment (Aleksanaa) [inline pkgs/by-name/ra/rainfuck-cli/package.nix:14]:
```
error: hash mismatch in fixed-output derivation '/nix/store/m49gyiip9bvzip8xj6l4515j4vnf5pnp-rainfuck-cli-1.0.0-vendor.tar.gz.drv':
         specified: sha256-n/ZWMEjSzyOBC9EVtGi9/13GnttCA5oqWlRcqan5gic=
            got:    sha256-nRMjRhrEuXbch7CMJII+VVXlUoC7f/B99dPKbqS3xQI=
```

**Comment by Aleksanaa** [comment]
Sorry for closing, CI not passing and without proper fix for a long time. If you want to work on this again, you can reopen it.

**Pull request closed without merging**